test/e2e/compact_test.go

```diff
@@ -768,7 +768,6 @@ func testCompactWithStoreGateway(t *testing.T, penaltyDedup bool) {
 		testutil.Ok(t, c.WaitSumMetricsWithOptions(e2emon.Greater(0), []string{"thanos_compact_iterations_total"}, e2emon.WaitMissingMetrics()))
 		testutil.Ok(t, c.WaitSumMetricsWithOptions(e2emon.Equals(18), []string{"thanos_compact_blocks_cleaned_total"}, e2emon.WaitMissingMetrics()))
 		testutil.Ok(t, c.WaitSumMetricsWithOptions(e2emon.Equals(0), []string{"thanos_compact_block_cleanup_failures_total"}, e2emon.WaitMissingMetrics()))
-		testutil.Ok(t, c.WaitSumMetricsWithOptions(e2emon.Equals(0), []string{"thanos_compact_blocks_marked_total"}, e2emon.WaitMissingMetrics()))
```

Comment:
AFAICT block cleaning (deletion) cannot happen without marking them first so I'm not sure how this even passed :sweat_smile:

Reply:
Is it because it is a different compactor?
Blocks are marked by the first compactor and deleted by the second compactor which is this one

Reply:
Mhm, it seems like you are correct. Maybe the first compactor sometimes doesn't mark all the blocks due to lag?

Reply:
> Maybe the first compactor sometimes doesn't mark all the blocks due to lag?

Yeah this sounds highly possible.

Reply:
With the race detector, the lag increases so at least on my machine it is constantly failing here. Also, it seems kind of pointless to assert that this metric is equal or greater than 0. Could we merge this, please?

Reply:
Sounds good!